lib/galaxy/managers/workflows.py

```diff
@@ -1972,3 +1978,9 @@ def import_workflow(self, workflow, **kwds):
         raise NotImplementedError(
             "Direct format 2 import of nested workflows is not yet implemented, use bioblend client."
         )
+
+
+def count_stored_workflow_user_assocs(session: Session, user, stored_workflow) -> int:
```

Comment:
Move into user class ?

Reply:
You mean the user manager class? I don't mind, although I feel we are querying workflows by user criteria here, not users by workflow criteria. If all queries like "get this and that stuff for this user" were located in the User model class, then yes; but I was hoping we could move such methods out of models and into [managers/repositories/or somewhere else] - see my comment on History.

Reply:
No, it's not a manager method. You can even set this up as a regular relationship if you want.

Reply:
Done.

In my opinion, just because one can setup a regular relationship for an associated model doesn't mean one should do that. You have a simple task: count the number of associations between `A` and `B`. A lightweight function that does just that and nothing more is the simplest solution here, I think.